src/module_2/analyzing_parquet_files.md

```diff
@@ -0,0 +1,576 @@
+    ----> 3 summary_segment_price = df_hyp2.groupby('user_segment')['price'].describe()
+          4 print(summary_segment_price)
+    
+
+    File c:\Users\ALEX\OneDrive\Escritorio\PROGRAMACION\ZRIVE\zrive-ds\.venv\Lib\site-packages\pandas\core\groupby\generic.py:1771, in DataFrameGroupBy.__getitem__(self, key)
+       1764 if isinstance(key, tuple) and len(key) > 1:
+       1765     # if len == 1, then it becomes a SeriesGroupBy and this is actually
+       1766     # valid syntax, so don't raise
+       1767     raise ValueError(
+       1768         "Cannot subset columns with a tuple with more than one element. "
+       1769         "Use a list instead."
+       1770     )
+    -> 1771 return super().__getitem__(key)
+    
+
+    File c:\Users\ALEX\OneDrive\Escritorio\PROGRAMACION\ZRIVE\zrive-ds\.venv\Lib\site-packages\pandas\core\base.py:244, in SelectionMixin.__getitem__(self, key)
+        242 else:
+        243     if key not in self.obj:
+    --> 244         raise KeyError(f"Column not found: {key}")
+        245     ndim = self.obj[key].ndim
+        246     return self._gotitem(key, ndim=ndim)
+    
+
+    KeyError: 'Column not found: price'
+
+
+### Question 1:
+Which products sell the most by geographic area, to see if there is variation in specific products between different areas.
+
+
+```python
+df_q1 = df_orders.merge(df_users[['user_id', 'user_nuts1']], on='user_id', how='left') 
+df_q1_expanded = df_q1.explode('ordered_items')
+df_q1_merged = df_q1_expanded.merge(df_inventory[['variant_id', 'product_type']], left_on='ordered_items', right_on='variant_id', how='left')
+df_q1_merged = df_q1_merged.dropna(subset=['product_type'])
+df_q1_merged.info()
+```
+
+    <class 'pandas.core.frame.DataFrame'>
+    Index: 92361 entries, 70 to 107957
+    Data columns (total 9 columns):
+     #   Column          Non-Null Count  Dtype         
+    ---  ------          --------------  -----         
+     0   id              92361 non-null  int64         
+     1   user_id         92361 non-null  object        
+     2   created_at      92361 non-null  datetime64[us]
+     3   order_date      92361 non-null  datetime64[us]
+     4   user_order_seq  92361 non-null  int64         
+     5   ordered_items   92361 non-null  object        
+     6   user_nuts1      91716 non-null  object        
+     7   variant_id      92361 non-null  float64       
+     8   product_type    92361 non-null  object        
+    dtypes: datetime64[us](2), float64(1), int64(2), object(4)
+    memory usage: 7.0+ MB
+    
+
+
+```python
+products_by_zone  = df_q1_merged.groupby(['user_nuts1', 'product_type']).size().unstack(fill_value=0)
+relative_products_by_zone = products_by_zone.div(products_by_zone.sum(axis=1), axis=0)
+```
+
+
+```python
+# Obtener los 3 productos más comunes por zona
+top_3_products_per_zone = relative_products_by_zone.apply(lambda row: row.nlargest(3).index.tolist(), axis=1)
+print(top_3_products_per_zone)
+```
+
+    user_nuts1
```

Comment:
I think it is also interesting to compute the relative number of sales per product for each nut. With this number, you are able to check which are the most sold products, but also if the proportion is similar or not between nuts.